sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1244,46 +1244,50 @@ case class Reverse(child: Expression) extends UnaryExpression with ImplicitCastI
   }
 
   private def arrayCodeGen(ctx: CodegenContext, ev: ExprCode, childName: String): String = {
-    val length = ctx.freshName("length")
-    val javaElementType = CodeGenerator.javaType(elementType)
+
     val isPrimitiveType = CodeGenerator.isPrimitiveType(elementType)
 
+    val numElements = ctx.freshName("numElements")
+    val arrayData = ctx.freshName("arrayData")
+
     val initialization = if (isPrimitiveType) {
-      s"$childName.copy()"
+      ctx.createUnsafeArray(arrayData, numElements, elementType, s" $prettyName failed.")
```

Comment:
It seems `UnsafeArrayData` is created per input row inside `createUnsafeArray`. Can't we avoid it?

Reply:
IIUC, the code in the master also create `UnsafeArrayData` per input row in `UnsafeArrayData.copy()`.

Reply:
You can check this? https://gist.github.com/maropu/e9e8afd64ce30cdf824bb4e18d0c9b4f

Reply:
I talked with @ueshin and no problem about this.